### PR TITLE
Let the DispReconstructor also compute a score for the sign prediction

### PR DESCRIPTION
**Pull request by maxnoe**

### CODE REVIEW

**Review by LukasBeiske — APPROVED** [review]

**Review by LukasBeiske — COMMENTED** [review]
Changelog missing

File: ctapipe/reco/sklearn.py [hunk 1]
```diff
@@ -681,12 +682,17 @@ def _predict(self, key, table):
             else:
                 prediction[valid] = valid_norms
 
-            prediction[valid] *= self._models[key][1].predict(X)
+            sign_proba = self._models[key][1].predict_proba(X)[:, 0]
+            # proba is [0 and 1] where 0 => very certain -1, 1 => very certain 1
```
Comment (Tobychev) [inline ctapipe/reco/sklearn.py:686]: An errant `-1`?

File: ctapipe/reco/sklearn.py [hunk 2]
```diff
@@ -945,7 +964,7 @@ def _cross_validate_regressor(self, telescope_type, train, test):
         prediction, _ = regressor._predict(telescope_type, test)
         truth = test[regressor.target]
         r2 = r2_score(truth, prediction)
-        return prediction, truth, {"R^2": r2}
+        return {f"{regressor.prefix}_energy": prediction}, truth, {"R^2": r2}
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply ctapipe/reco/sklearn.py:967]: There is no direct test, yes.

However the code is covered by the tests for the train Tool, which also inspect the cross validation output.

**Review by Tobychev — APPROVED** [review]

**Comment by maxnoe** [comment]
Here a plot of the distribution for LST in prod5 (only 50k events, simple model config):

disp_score

Looks as expected, I would say

**Pull request merged**